live_validation_scripts/validate_live_pixel.mjs

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env node
+
+import csv from 'csv-parser';
+import fs from 'fs';
+
+import { getArgParserWithCsv } from '../src/args_utils.mjs';
+import { ParamsValidator } from '../src/params_validator.mjs';
+import { LivePixelsValidator } from '../src/live_pixel_validator.mjs';
+
+import * as fileUtils from '../src/file_utils.mjs';
+
+const argv = getArgParserWithCsv('Validates pixels from the provided CSV file', 'path to CSV file containing pixels to validate').parse();
+
+function main(mainDir, csvFile) {
+    console.log(`Validating live pixels in ${csvFile} against definitions from ${mainDir}`);
+
+    const productDef = fileUtils.getProductDef(mainDir);
+    const forceLowerCase = productDef.forceLowerCase;
+
+    const commonParams = fileUtils.getCommonParams(mainDir, forceLowerCase);
+    const commonSuffixes = fileUtils.getCommonSuffixes(mainDir, forceLowerCase);
+
+    const tokenizedPixels = fileUtils.getTokenizedPixels(mainDir, forceLowerCase);
+    const paramsValidator = new ParamsValidator(commonParams, commonSuffixes);
+    const ignoreParams = fileUtils.getIgnoreParams(mainDir, forceLowerCase);
+
+    const liveValidator = new LivePixelsValidator(tokenizedPixels, productDef, ignoreParams, paramsValidator);
+    let processedPixels = 0;
+    fs.createReadStream(csvFile)
+        .pipe(csv())
+        .on('data', (row) => {
+            processedPixels++;
+            if (processedPixels % 100000 === 0) {
+                console.log(`...Processing row ${processedPixels.toLocaleString('en-US')}...`);
+            }
+            liveValidator.validatePixel(row.pixel, row.params);
+        })
```

Comment:
Could we have the option to read from `stdin`? This would allow streaming the data directly from clickhouse to validation, without needing to store data on disk in between.

Reply:
We can, but I worry what happens when we are piping large amounts of data like that. Also, I think it's still good to have data on disk: 
* If something breaks, we can re-run the validation without having to re-fetch the data
* We have an estimate of how long validation will take based on number of lines in the CSV

Eventually, once this is running on Jenkins somewhere and all intermediary processing will be destroyed anyway, it may make sense to switch to piping, but I don't think we're there yet.